### PR TITLE
Update tagging terminology

### PR DESCRIPTION
**Pull request by hannako**
## What 

This PR completes the changes to tagging terminology in our publishing apps. Related work:
- [collections publisher](https://github.com/alphagov/collections-publisher/pull/1582)
- [content tagger](https://github.com/alphagov/content-tagger/pull/1307)

To note:

- Some commits have failing e2e tests - these will pass once https://github.com/alphagov/publishing-e2e-tests/pull/489/files is merged.
- [This](https://docs.google.com/spreadsheets/d/1OLw4B46ZqTgsuo762Kjxb-M6_rfX_dW23qp8bNhZrMo/edit#gid=373458138) is the content changes document that the commits refer to.

## Why

Clear consistent language in the publishing apps will help publishers tag
content better, and understand the implications of this. Better tagging will
make their content more findable.

## Trello

https://trello.com/c/Sbws55cL/928-whitehall-update-copy-based-on-jenns-doc-m

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: test/functional/admin/publications_controller_test.rb [hunk 1]
```diff
@@ -218,7 +218,7 @@ class Admin::PublicationsControllerTest < ActionController::TestCase
     get :show, params: { id: publication }
 
     assert_selected_specialist_sectors_are_displayed
-    assert_select "a[href='#{edit_admin_edition_legacy_associations_path(publication)}']", /Change Associations/
+    assert_select "a[href='#{edit_admin_edition_legacy_associations_path(publication)}']", /Change specialist topic tags/
```
Comment (chao-xian) [inline test/functional/admin/publications_controller_test.rb:221]: I'm just wondering whether it's worth renaming the controller and route for this... thoughts? https://github.com/alphagov/whitehall/blob/main/app/controllers/admin/edition_legacy_associations_controller.rb (plus template, test and references to it)

Comment (chao-xian) [reply test/functional/admin/publications_controller_test.rb:221]: And this features/support/admin_legacy_associations_helper.rb

**Review by chao-xian — APPROVED** [review]
Lovely stuff @hannako! So easy to review and sensible changes. Just a comment on whether we should be really consistent and tidy and rename controllers/routes/etc to not be "legacy associations" as well for you to consider. Maybe as a separate PR?

File: test/functional/admin/publications_controller_test.rb (hunk 1 above)
Comment (hannako) [reply test/functional/admin/publications_controller_test.rb:221]: Yes I totally agree - there's a piece of work in all three apps following these label change PR's ([content tagger](https://github.com/alphagov/content-tagger/pull/1307), [collections-publisher](https://github.com/alphagov/collections-publisher/pull/1582))to sync the back end naming with the new frontend terminology. I thooooooought there was a card for that, but I can spot it in the backlog so I'll make one.

**Pull request merged**